WalletWasabi.Gui/Controls/WalletExplorer/SendControlViewModel.cs

```diff
@@ -945,6 +923,12 @@ public override void OnOpen(CompositeDisposable disposables)
 				.ObserveOn(RxApp.MainThreadScheduler)
 				.Subscribe(_ => IsSliderFeeUsed = true);
 
+			Global.UiConfig.WhenAnyValue(x => x.FeeDisplayFormat)
+				.ObserveOn(RxApp.MainThreadScheduler)
+				.Subscribe(format =>
+					FeeDisplayFormat = (FeeDisplayFormat)(Enum.ToObject(typeof(FeeDisplayFormat), format) ?? FeeDisplayFormat.SatoshiPerByte))
```

Comment:
I think this is a counterexample that breaks your implementation as `(FeeDisplayFormat)(Enum.ToObject(typeof(FeeDisplayFormat)` does not return `null` (normally casting this way throws an exception, which would be even worse but that's not what happens in this case):

```csharp
using System;

public class Test
{
	public enum FeeDisplayFormat
	{
		USD,
		BTC,
		SatoshiPerByte,
		Percentage
	}
	
	public static void Main()
	{
		Console.WriteLine((FeeDisplayFormat)1); // Prints "BTC"
		Console.WriteLine((FeeDisplayFormat)10); // Prints "10" ... kind of unexpected
		Console.WriteLine((FeeDisplayFormat)Enum.ToObject(typeof(FeeDisplayFormat), 10)); // Prints "10"
		Console.WriteLine((FeeDisplayFormat)(Enum.ToObject(typeof(FeeDisplayFormat), 10) ?? FeeDisplayFormat.SatoshiPerByte)); // Prints "10"
	}
}
```

https://ideone.com/RlQG5T

`FeeDisplayFormat = (FeeDisplayFormat)(Enum.ToObject(typeof(FeeDisplayFormat), format) ?? FeeDisplayFormat.SatoshiPerByte)` <- I don't thing that those `??` ever do something.

https://stackoverflow.com/questions/29482/how-can-i-cast-int-to-enum 

I would use something like this:

```csharp
if (Enum.IsDefined(typeof(MyEnum), format)) { 
   return (MyEnum)format; 
} else { 
  return FeeDisplayFormat.SatoshiPerByte; 
}
```

Reply:
Fixed